api/idp/api.go

```diff
@@ -8,56 +8,62 @@ import (
 
 // IdentityProviderRepresentation struct
 type IdentityProviderRepresentation struct {
-	AddReadTokenRoleOnCreate  *bool                   `json:"addReadTokenRoleOnCreate,omitempty"`
-	Alias                     *string                 `json:"alias,omitempty"`
-	AuthenticateByDefault     *bool                   `json:"authenticateByDefault,omitempty"`
-	Config                    *map[string]interface{} `json:"config,omitempty"`
-	DisplayName               *string                 `json:"displayName,omitempty"`
-	Enabled                   *bool                   `json:"enabled,omitempty"`
-	FirstBrokerLoginFlowAlias *string                 `json:"firstBrokerLoginFlowAlias,omitempty"`
-	InternalID                *string                 `json:"internalId,omitempty"`
-	LinkOnly                  *bool                   `json:"linkOnly,omitempty"`
-	PostBrokerLoginFlowAlias  *string                 `json:"postBrokerLoginFlowAlias,omitempty"`
-	ProviderID                *string                 `json:"providerId,omitempty"`
-	StoreToken                *bool                   `json:"storeToken,omitempty"`
-	TrustEmail                *bool                   `json:"trustEmail,omitempty"`
+	AddReadTokenRoleOnCreate    *bool                   `json:"addReadTokenRoleOnCreate,omitempty"`
+	Alias                       *string                 `json:"alias,omitempty"`
+	AuthenticateByDefault       *bool                   `json:"authenticateByDefault,omitempty"`
+	Config                      *map[string]interface{} `json:"config,omitempty"`
+	DisplayName                 *string                 `json:"displayName,omitempty"`
+	Enabled                     *bool                   `json:"enabled,omitempty"`
+	FirstBrokerLoginFlowAlias   *string                 `json:"firstBrokerLoginFlowAlias,omitempty"`
+	HideOnLogin                 *bool                   `json:"hideOnLogin,omitempty"`
+	InternalID                  *string                 `json:"internalId,omitempty"`
+	LinkOnly                    *bool                   `json:"linkOnly,omitempty"`
+	PostBrokerLoginFlowAlias    *string                 `json:"postBrokerLoginFlowAlias,omitempty"`
+	ProviderID                  *string                 `json:"providerId,omitempty"`
+	StoreToken                  *bool                   `json:"storeToken,omitempty"`
+	TrustEmail                  *bool                   `json:"trustEmail,omitempty"`
+	UpdateProfileFirstLoginMode *string                 `json:"updateProfileFirstLoginMode,omitempty"`
```

Comment:
Are you considering values on/off rather than true/false?

Reply:
Yes, the two expected values for this are "on"/"missing"/"off", and in our [template ](https://projectportal.elca.ch/jira/browse/CLOUDTRUST-6568) we set it to "on" for all our IDPs. But actually, this attribute is marked as deprecated in the [docs ](https://www.keycloak.org/docs-api/latest/javadocs/org/keycloak/representations/idm/IdentityProviderRepresentation.html#updateProfileFirstLoginMode)... So with @harture we decided to remove this attribute for now